packages/kbn-apm-synthtrace/src/lib/logs/logs_synthtrace_es_client.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { dedot, ESDocumentWithOperation } from '@kbn/apm-synthtrace-client';
+import { pipeline, Readable, Transform } from 'stream';
+import { LogDocument } from '@kbn/apm-synthtrace-client/src/lib/logs';
+import { SynthtraceEsClient, SynthtraceEsClientOptions } from '../shared/base_client';
+import { getSerializeTransform } from '../shared/get_serialize_transform';
+import { Logger } from '../utils/create_logger';
+
+export type LogsSynthtraceEsClientOptions = Omit<SynthtraceEsClientOptions, 'pipeline'>;
+
+export class LogsSynthtraceEsClient extends SynthtraceEsClient<LogDocument> {
+  constructor(options: { client: Client; logger: Logger } & LogsSynthtraceEsClientOptions) {
+    super({
+      ...options,
+      pipeline: logsPipeline(),
+    });
+    this.dataStreams = ['logs-*-*'];
+  }
+}
+
+function logsPipeline() {
+  return (base: Readable) => {
+    return pipeline(
+      base,
+      getSerializeTransform<LogDocument>(),
+      getRoutingTransform(),
+      getDedotTransform(),
+      (err: unknown) => {
+        if (err) {
+          throw err;
+        }
+      }
+    );
+  };
+}
+
+function getRoutingTransform() {
+  return new Transform({
+    objectMode: true,
+    transform(document: ESDocumentWithOperation<LogDocument>, encoding, callback) {
+      if (
+        'data_stream.type' in document &&
+        'data_stream.dataset' in document &&
+        'data_stream.namespace' in document
```

Comment:
Could we include a default namespace? Most of the times people would ingest logs in `default` namespace
Something like 
```
document['data_stream.namespace'] ?? 'default'
```

Reply:
The Logs client takes care of it. It defaults the value like this

https://github.com/achyutjhunjhunwala/kibana/blob/86907222272e8f3cf25496664d776643cf5165e4/packages/kbn-apm-synthtrace-client/src/lib/logs/index.ts#L67-L69

This is just an extra a check